File: WOLF/wolf_b/app/db/models/agent.py
from sqlalchemy import Column, String, Text, DateTime, Enum
from sqlalchemy.sql import func
from app.db.database import Base
import enum

class AgentStatus(str, enum.Enum):
    IDLE = "idle"
    ANALYZING = "analyzing"
    WORKING = "working"
    WAITING = "waiting"
    ERROR = "error"
    COMPLETED = "completed"

class Agent(Base):
    __tablename__ = "agents"

    id = Column(String, primary_key=True, index=True)
    role = Column(String, unique=True, nullable=False, index=True)
    name = Column(String, nullable=False)
    description = Column(Text)
    system_prompt = Column(Text)
    status = Column(String, default=AgentStatus.IDLE.value)
    current_task = Column(String)
    capabilities = Column(Text)  # JSON string
    llm_provider = Column(String)
    model = Column(String)
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), onupdate=func.now())

File: WOLF/wolf_f/src/types/agent.ts
// Agent Types
export type AgentRole =
  | 'main'          // Main Agent (Claude Code style)
  | 'pm'           // Project Manager
  | 'research'     // Research Agent
  | 'ml'           // ML Engineer
  | 'developer'     // Full-Stack Developer
  | 'writer'        // Technical Writer
  | 'data'          // Data Engineer
  | 'review'        // Review Agent
  | 'devops';       // DevOps Agent

export type AgentStatus =
  | 'idle'
  | 'analyzing'
  | 'working'
  | 'waiting'
  | 'error'
  | 'completed';

export interface AgentConfig {
  id: string;
  role: AgentRole;
  name: string;
  description: string;
  systemPrompt: string;
  status: AgentStatus;
  currentTask?: string;
  capabilities: string[];
  llmProvider?: string;
  model?: string;
}

export interface AgentMessage {
  id: string;
  from: AgentRole;
  to: AgentRole | 'broadcast';
  type: 'task' | 'result' | 'question' | 'approval' | 'rejection' | 'status' | 'progress' | 'broadcast';
  content: string;
  attachments?: Attachment[];
  metadata?: Record<string, unknown>;
  timestamp: number;
  taskId?: string;
  sessionId?: string;
}

export interface Attachment {
  id: string;
  name: string;
  type: string;
  url?: string;
  content?: string;
}

// Agent State
export interface AgentState {
  agents: Map<string, AgentConfig>;
  activeAgent: string | null;
  agentMessages: Map<string, AgentMessage[]>;
}
